.github/workflows/zerodowntime_tests.yml

```
@@ -0,0 +1,453 @@
+          fi
+          
+          KEPTN_API_TOKEN=$(kubectl get secret keptn-api-token -n ${KEPTN_NAMESPACE} -ojsonpath={.data.keptn-api-token} | base64 --decode)
+
+          echo "KEPTN_ENDPOINT=${KEPTN_ENDPOINT}"
+          echo "##[set-output name=KEPTN_ENDPOINT;]$(echo ${KEPTN_ENDPOINT})"
+
+      - name: Install Helm/JMeter Service
+        timeout-minutes: 5
+        env:
+          KEPTN_ENDPOINT: ${{ steps.determine_keptn_endpoint.outputs.KEPTN_ENDPOINT }}
+          HELM_SERVICE_HELM_CHART_NAME: ${{ steps.extract_helm_chart_name.outputs.HELM_SERVICE_HELM_CHART_NAME }}
+          JMETER_SERVICE_HELM_CHART_NAME: ${{ steps.extract_helm_chart_name.outputs.JMETER_SERVICE_HELM_CHART_NAME }}
+        run: |
+      
+          # In-cluster execution plane
+          helm install helm-service "${HELM_SERVICE_HELM_CHART_NAME}" -n ${{ env.KEPTN_NAMESPACE }}
+          helm install jmeter-service "${JMETER_SERVICE_HELM_CHART_NAME}" -n ${{ env.KEPTN_NAMESPACE }}
+        
+          helm test jmeter-service -n ${{ env.KEPTN_NAMESPACE }}
+          helm test helm-service -n ${{ env.KEPTN_NAMESPACE }}
+        
+
+      - name: Prepare test run
+        id: prepare_test_run
+        run: |
+          go install gotest.tools/gotestsum@v1.7.0
+          cd test/zerodowntime
+          go get ./...
+
+      - name: "------- TESTS START -------"
+        run: echo "------- TESTS START -------"
+
+      #######################################################################
+      # TESTS
+      #######################################################################
+
+      #######################################################################
+      # COMMON TESTS
+      ##############
+
+      - name: Zero Downtime Tests
+        id: test_aggregated
+        timeout-minutes: 90
+        env:
+          KEPTN_ENDPOINT: ${{ steps.determine_keptn_endpoint.outputs.KEPTN_ENDPOINT }}
+          DO_AUTH: "true"
+          UNLEASH_SERVICE_VERSION: "release-0.3.2"
+          GOTESTSUM_FORMAT: "testname"
+        run: |
+          cd test/zerodowntime          
+          gotestsum --no-color=false -- -run Test_ZeroDowntime -v -timeout 90m
+
+      - name: "------- TESTS END -------"
+        run: echo "------- TESTS END -------"
+
+      #######################################################################
+      # TEARDOWN
+      #######################################################################
+
+      - name: keptn generate support-archive
+        if: always()
+        env:
+          SUPPORT_ARCHIVE_FILENAME: keptn-support-archive-${{ github.run_id }}-${{ matrix.PLATFORM }}-${{ matrix.CLOUD_PROVIDER}}-${{ matrix.PLATFORM_VERSION }}
+        timeout-minutes: 5
+        run: |
+          mkdir support-archive/
+          echo "y" | keptn generate support-archive -n ${KEPTN_NAMESPACE}
+          mv support-archive/keptn*.zip support-archive/${SUPPORT_ARCHIVE_FILENAME}.zip
+
+      - name: Cleanup test namespace
```

```suggestion
      - name: Cleanup test namespace
        if: always()
```

this will ensure that even if some step before this fails, it's still executed